service/header/store.go

```diff
@@ -140,10 +130,21 @@ func (s *store) Stop(ctx context.Context) error {
 
 func (s *store) Head(ctx context.Context) (*ExtendedHeader, error) {
 	head, err := s.GetByHeight(ctx, s.heightSub.Height())
-	if err == ErrNotFound {
+	if err == nil {
+		return head, err
```

Comment:
```suggestion
		return head, nil
```

Reply:
We that it is nil by the if case, but it doesn't hurt to say it is nil explicitly.